### PR TITLE
Borg Jetpack Overhaul

### PR DESCRIPTION
**Pull request by NanakoAC**
Somewhat rewrote the cyborg jetpack to be an internal component instead
of a software upgrade. this allows it to be removed or destroyed, and to
work without putting something in the borg's inventory.
Also fixed it, it was broken.
Cyborg jetpack can now be installed on a wider array of borgs that have
reason to go EVA: Crisis, security, engineering, construction (also the
unused combat borg, incase that gets re-enabled)

### CODE REVIEW

File: Nanako-PR-341.txt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#   maptweak
+#   spellcheck (typo fixes)
+#   experiment
+#################################
+
+# Your name.  
+author: Nanako
+
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: False
```
Comment (Arrow768) [inline Nanako-PR-341.txt]: Shouldn´t this be True ?

Comment (NanakoAC) [inline Nanako-PR-341.txt]: Perhaps i'm using it wrong, my understanding was that the delete-after was for whether it's deleted from my computer, an i'd like to keep them. 
if this isn't the case, ill set them true in future

Comment (Arrow768) [inline Nanako-PR-341.txt]: You would set delete-after to false if you want to come back later in the future (in a additional merge request) and change the contents of this changelog.

Comment (skull132) [inline Nanako-PR-341.txt]: @Arrow768 is right. This should be True. DNM until changed.

File: code/modules/mob/living/silicon/robot/component.dm [hunk 2]
```diff
@@ -189,6 +214,11 @@
 	components["camera"] = new/datum/robot_component/camera(src)
 	components["comms"] = new/datum/robot_component/binary_communication(src)
 	components["armour"] = new/datum/robot_component/armour(src)
+	components["jetpack"] = new/datum/robot_component/jetpack(src)
+	var/datum/robot_component/JC = components["jetpack"]
+	world << "Setting installed of Jetpack component: [JC] [JC.installed]"
```
Comment (skull132) [inline code/modules/mob/living/silicon/robot/component.dm]: Debug messages in pushed code. Need to be removed.

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 3]
```diff
@@ -70,6 +70,7 @@
 	var/modtype = "Default"
 	var/lower_mod = 0
 	var/jetpack = 0
```
Comment (skull132) [inline code/modules/mob/living/silicon/robot/robot.dm]: If this is a boolean, and all code using it does't do much beyond `if (jetpack)` or what have you, then this can be edited to simply be `var/obj/item/weapon/tank/jetpack/carbondioxide/synthetic/jetpack = null` and you can get rid of `jetpackRef`.

An object variable having an ingame referenced assigned returns the same way a boolean set to true does. And if it's null, with no reference, it'll return the same way a boolean set to negative does. 99% of the time, this can be used.

**Comment by NanakoAC** [comment]
This is finished and ready to merge now

File: code/modules/mob/living/silicon/robot/component.dm [hunk 4]
```diff
@@ -183,12 +211,17 @@
...
 	components["radio"] = new/datum/robot_component/radio(src)
 	components["power cell"] = new/datum/robot_component/cell(src)
 	components["diagnosis unit"] = new/datum/robot_component/diagnosis_unit(src)
 	components["camera"] = new/datum/robot_component/camera(src)
 	components["comms"] = new/datum/robot_component/binary_communication(src)
 	components["armour"] = new/datum/robot_component/armour(src)
+	components["jetpack"] = new/datum/robot_component/jetpack(src)
+	jetpackComponent = components["jetpack"]
+	var/datum/robot_component/JC = components["jetpack"]
+	JC.installed = 0//We start the jetpack as not installed, because its nondefault
```
Comment (skull132) [inline code/modules/mob/living/silicon/robot/component.dm]: Why is this necessary, I'm curious? It technically installs it, but then changes a variable to make it "appear" not installed. While still being there?

Why not just leave it uninstalled?

Comment (NanakoAC) [inline code/modules/mob/living/silicon/robot/component.dm]: This is the way all components work. Components are never removed from the components list, they're simply toggled to uninstalled when removed, and spawn the external item. Changing it would be lots of work for little benefit, to remove various reference errors that would result, and making the jetpack be a wierd exception runs counter to why i did this in the first place

Comment (skull132) [inline code/modules/mob/living/silicon/robot/component.dm]: That's dumb, but sure, since the rest works that way as well.

Though, since speed is what you were interested in a while back. `var/jetpackComponent` is already instantiated as a `datum/robot_component`. Ergo, it has the `installed` variable. Instead of instantiating `JC`, it might be an idea to simply do:
```
	components["jetpack"] = new/datum/robot_component/jetpack(src)
	jetpackComponent = components["jetpack"]
	jetpackComponent.installed = 0 //Because raisins.
```

**Comment by NanakoAC** [comment]
that little thing is done.

FWIW, i think of the internal component, if not installed, as a mounting point for that component
anyway this is definitely totally ready
